pkg/i2gw/providers/nginx/annotations/hsts.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"strconv"
+	"strings"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/intermediate"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/notifications"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common"
+)
+
+// HSTSFeature converts HSTS annotations to HTTPRoute ResponseHeaderModifier filters.
+// Supports nginx.org/hsts, nginx.org/hsts-max-age, and nginx.org/hsts-include-subdomains annotations.
+func HSTSFeature(ingresses []networkingv1.Ingress, _ map[types.NamespacedName]map[string]int32, ir *intermediate.IR) field.ErrorList {
+	var errs field.ErrorList
+
+	ruleGroups := common.GetRuleGroups(ingresses)
+	for _, rg := range ruleGroups {
+		for _, rule := range rg.Rules {
+			if hsts, ok := rule.Ingress.Annotations[nginxHSTSAnnotation]; ok && hsts == "true" {
+				errs = append(errs, processHSTSAnnotation(rule.Ingress, ir)...)
+			}
+		}
+	}
+
+	return errs
+}
+
+//nolint:unparam // ErrorList return type maintained for consistency
+func processHSTSAnnotation(ingress networkingv1.Ingress, ir *intermediate.IR) field.ErrorList {
+	var errs field.ErrorList
```

Comment:
Is this not needed? Its added in the other functions.

```suggestion
	var errs field.ErrorList //nolint:unparam // ErrorList return type maintained for consistency
```

Reply:
I think I'm not returning any errors but let me confirm